include/aws/io/retry_strategy.h

```diff
@@ -108,8 +113,20 @@ struct aws_exponential_backoff_retry_options {
     /** Jitter mode to use, see comments for aws_exponential_backoff_jitter_mode.
      * Default is AWS_EXPONENTIAL_BACKOFF_JITTER_DEFAULT */
     enum aws_exponential_backoff_jitter_mode jitter_mode;
-    /** By default this will be set to use aws_device_random. If you want something else, set it here. */
+
+    /** Deprecated. Use generate_random_impl instead
+     * By default this will be set to use aws_device_random. If you want something else, set it here.
+     * */
     uint64_t (*generate_random)(void);
+
+    /*
+     * By default this will be set to use aws_device_random. If you want something else, set it here.
+     */
+    generate_random_fn *generate_random_impl;
+    /**
+     * Optional user data for the generate random generate_random_impl.
+     */
+    void *generate_random_user_data;
```

Comment:
I'd like to see the code that's using this.

I'm worried that ... now that we're passing a `void *` from another language to C. that other language is going to need to know when C is done using that `void *`. Like, do we need to add a shutdown callback too?

Reply:
Thanks, that is a great catch. I have added a shutdown callback and relevant changes in Swift at https://github.com/awslabs/aws-crt-swift/pull/147/files